src/scripts/upgrades/Upgrade.ts

```diff
@@ -0,0 +1,79 @@
+///<reference path="../GameConstants.ts"/>
+///<reference path="Cost.ts"/>
+///<reference path="CostFactory.ts"/>
+
+class Upgrade {
+    name: any;
+    displayName: string;
+    maxLevel: number;
+    _level: KnockoutObservable<number> = ko.observable();
+
+    // Describes whether this upgrade increases or decreases a number.
+    // (e.g. power is increasing, time is decreasing).
+    increasing: boolean;
```

Comment:
`Upgrade.increasing` doesn't appear to be used anywhere. The only place where it might be used `Underground.getEnergyRegenTime`; but these underground upgrades are hard-coded to be either + or - the benefit. Is `increasing` redundant? Even if we have a later upgrade that doesn't hard code the direction of the math, we can still have `Upgrade.bonusList` be negative numbers, right?

Reply:
It is used in the ui, to display a + or - sign. My idea was to have bonuses always be positive for consistency, when a value gets calculated the bonus gets interpret by that function as you see in the energyRegenTime.

Reply:
If the function interpretation and the display are consistent, then wouldn't it be more concise to have `current_value + bonus = new_value` at all times, even if bonus is negative? If the word bonus gives the connotation of something positive, can change it to something else, like "effect".